### ISSUE TITLE
As a front end user, I want to filter search results to records with transcription available, so that I can easily find documents that have already been transcribed and will be easier for me to use.

### ISSUE THREAD

**Issue by blms** [Princeton-CDH/geniza#630]
## testing notes (qa)

- Navigate to the document search page on the QA site, type in a search query and press the submit button
- [x] Click the "Filters" button below the search bar and confirm "Has Transcription" appears as a filter with a count of documents matching your search terms that have transcriptions
- [ ] Enable that filter and click "Apply", then confirm the results all have transcriptions

**Comment by rlskoeser** [comment]
@blms I want to split this out into multiple stories; does it make sense to use this one for the ui/ux portion of the modal for filters and add other stories for the other features? (it's ok if you work on them in tandem, but they should each be tested separately)

**Comment by blms** [comment]
@rlskoeser That makes sense to me! Would it be ok to use this one for the solr fields/indexing, since I referenced it in that one related commit on Friday?

**Comment by blms** [comment]
@rlskoeser Went ahead and made https://github.com/Princeton-CDH/geniza/issues/633 for the filter modal, hope that is ok!

**Comment by rlskoeser** [comment]
@blms yes, that works! thank you

**Comment by richmanrachel** [comment]
@blms and @rlskoeser - a few things not working as expected:
1) When I click the button for filters, it takes me to a whole separate page (rather than expanding within the search page)
2) The filters on that second page did not adjust immediately to the search term I entered before clicking it, so it was giving me the number of documents with transcriptions from the whole site, not just my search term. 
![image](https://user-images.githubusercontent.com/68028846/155750661-4df95526-e2ca-40d7-b661-037336afefce.png)

3) I think the "has transcription" filter is correctly pulling content, but there is one document that doesn't have proper content (and is probably an issue with the TEI because it appears the same way in the admin site): 
![image](https://user-images.githubusercontent.com/68028846/155750409-ada8b636-317b-4a0a-b9ce-15d57bd926ff.png)

**Comment by richmanrachel** [comment]
I guess no. 1 is just a style choice, but it's weird to not see your search term anymore. And it's compounded by no. 2 to be disorienting, because if you're navigated to a new page, you expect changes to have already been made.

**Comment by richmanrachel** [comment]
4) I just tried adding the filter for transcription with no search term on random sort, and no transcriptions are appearing. I'm not sure what the best solution here is because I imagine there are competing codes at stake, but it would be nice to be able to just look at a list of all the documents with transcriptions without having to enter a specific search term.

**Comment by blms** [comment]
@richmanrachel Thanks for the thorough test!

1. That's indeed a design decision, @gissoo thoughts?
2. I agree that's confusing, I believe we are planning on updating those counts when you write your search term. For now, you'll have to submit the search before the filter counts will update. (Looks like Rebecca just mentioned this in #636)
3. I think that's an unrelated issue with data, I'll let @rlskoeser weigh in!
4. I can't replicate that. Are you sure it was selected and not deselected? It's very subtle right now but green is selected and black is deselected. Gissoo is working on this in #650.

**Comment by rlskoeser** [comment]
Yes, 3 seems to be a data issue related to content that looks like transcription but isn't — Alan mentioned something similar in Slack. I'm not sure where the content is coming from, but we should investigate and address.  I'll make a new issue from your comment.

**Comment by richmanrachel** [comment]
@blms - regarding 4, it seems that the URL is capturing what is selected, but there's no indication to the user otherwise. I don't see any green or black differentiation: 
![image](https://user-images.githubusercontent.com/68028846/155754707-f510d7a4-45db-4e7f-8098-8044d9d875b4.png)

**Comment by blms** [comment]
Those are all green! You can kind of tell if you compare with the down arrow for "Document Type".

<img width="83" alt="Screen Shot 2022-02-25 at 11 55 41 AM" src="https://user-images.githubusercontent.com/7234006/155755493-3f7f3aeb-700a-44fe-b208-011aba64de4c.png">

That means you should only see documents that have all three transcription, translation, and discussion.

**Comment by richmanrachel** [comment]
@blms - haha, glad it's in the works. I usually don't think of myself as color blind, but I really can't see it until zooming in! 

Should we close this issue because the dev works, but track decision making for design about where the filters show up elsewhere?

**Comment by rlskoeser** [comment]
yes, let's close since the functionality is working correctly

**Comment by gissoo** [comment]
@richmanrachel Correct, that is the same design we agreed on earlier last year. The reason that the filters open on a different page is that in the future they will clutter the page and push down the results further whenever they are expanded. But this is worth to find out about through usability testing.

And the issue you described with the search I believe is happening because we don't have the reactive search yet so you have to click enter (I see it was mentioned that you discussed it in #636, I'm glad it will be resolved, you're right @richmanrachel it's not intuitive right now). 

However, your comments make me wonder if it would be worth adding the search bar to the filters window, so that when you go the filters window you could still see your search term in the search box – do you have any thoughts @rlskoeser @blms ?

**Comment by rlskoeser** [comment]
Maybe I wasn't paying close enough attention to the designs on Figma, but it wasn't obvious to me that the filters would be full screen until I saw them implemented. That makes some sense to me for mobile — I think it's a common enough pattern — but it bothers me on desktop.  Here are a few reasons why I think it's bothering me:

1. the fact that Rachel thinks she's on a separate page, instead of a modal or dialog on the same page; this seems disorienting and unhelpful
2. after clicking on the filters toggle button, I have to move my mouse a fair bit to get to where the filters are and apply them — I would expect them to be readily available near where I clicked to turn them on.

@gissoo we could probably include the search bar in the filters window, but wouldn't it be better to expand the filters in place where they are (absolutely positioned, on top of the form below) so you have the context of your search preserved?
